src/transport/MixTransport.cpp

```diff
@@ -50,8 +50,80 @@ double MixTransport::thermalConductivity()
 
 void MixTransport::getThermalDiffCoeffs(double* const dt)
 {
-    for (size_t k = 0; k < m_nsp; k++) {
-        dt[k] = 0.0;
+    update_T();
+    update_C();
+    if (!m_bindiff_ok) {
+        updateDiff_T();
+    }
+    if (!m_viscwt_ok) {
+        updateViscosity_T();
+    }
+
+    double mmw = m_thermo->meanMolecularWeight();
+    const std::vector<double>& M = m_thermo->molecularWeights();
+    const double* y = m_thermo->massFractions();
+
+    std::vector<double>& a = m_spwork;
+
+    for (size_t k=0; k<m_nsp; ++k) {
+        dt[k] = 0.;
+        double xk = y[k]*mmw/M[k];
```

Comment:
Would it be faster to precompute this value for all of the species? I suppose it would use a little extra memory, but especially since the mole fraction is computed again in the nested loop, it seems worth it